integration_tests/src/main/python/arithmetic_ops_test.py

```diff
@@ -535,3 +535,35 @@ def test_div_by_zero_ansi(expr):
 @pytest.mark.parametrize('expr', ['1/0', 'a/0', 'a/b'])
 def test_div_by_zero_nonansi(expr):
     _test_div_by_zero(ansi_mode='nonAnsi', expr=expr)
+
+
+def _test_div_overflow(ansi_mode, expr):
+    ansi_conf = {'spark.sql.ansi.enabled': ansi_mode == 'ansi'}
+    data_gen = lambda spark: two_col_df(
+        spark,
+        LongGen(max_val=LONG_MIN),
+        IntegerGen(min_val=-1, max_val=-1), length=1)
+    div_overflow_func = lambda spark: data_gen(spark).selectExpr(expr)
+
+    if ansi_mode == 'ansi':
+        assert_gpu_and_cpu_error(
+            df_fun=lambda spark: div_overflow_func(spark).collect(),
+            conf=ansi_conf,
+            error_message='java.lang.ArithmeticException: Overflow in integral divide')
+    else:
+        assert_gpu_and_cpu_are_equal_collect(div_overflow_func, ansi_conf)
+
+div_overflow_exprs = [
+    'CAST(-9223372036854775808L as LONG) DIV -1',
+    'a DIV CAST(-1 AS INT)',
+    'a DIV b']
+
+@pytest.mark.parametrize('expr', div_overflow_exprs)
+@pytest.mark.xfail(condition=is_before_spark_320(), reason='https://github.com/apache/spark/pull/32260')
+def test_div_overflow_ansi(expr):
+    _test_div_overflow(ansi_mode='ansi', expr=expr)
+
+
+@pytest.mark.parametrize('expr', div_overflow_exprs)
+def test_div_overflow_nonansi(expr):
```

Comment:
Why not just have a second boolean parameter for ANSI vs not-ANSI instead of a separate test with the same implementation?

Reply:
Thanks for review. Updated the tests.